In [1]:
import pandas as pd
import numpy as np
import random
import io

### Leer la base de datos

In [2]:
from google.colab import files
uploaded = files.upload()

Saving econdata.csv to econdata.csv


In [3]:
econdata = pd.read_csv(io.BytesIO(uploaded["econdata.csv"]))

In [4]:
econdata.head()

id                  geo_point_2d  \
0   0   19.424781053,-99.1327537959   
1   1  19.4346139576,-99.1413808393   
2   2  19.4340695945,-99.1306348409   
3   3      19.42489472,-99.12073393   
4   4      19.42358238,-99.12451093   

                                           geo_shape   clave_cat  \
0  {"type": "Polygon", "coordinates": [[[-99.1332...  307_130_11   
1  {"type": "MultiPoint", "coordinates": [[-99.14...  002_008_01   
2  {"type": "MultiPoint", "coordinates": [[-99.13...  006_002_12   
3  {"type": "MultiPoint", "coordinates": [[-99.12...  323_102_06   
4  {"type": "MultiPoint", "coordinates": [[-99.12...  323_115_12   

            delegacion perimetro     tipo  \
0           Cuauhtémoc         B  Mercado   
1            Cuautémoc         A    Museo   
2            Cuautémoc         A    Museo   
3  Venustiano Carranza         B    Hotel   
4  Venustiano Carranza         B    Hotel   

                                              nom_id  
0                                        Pino Suárez  
1  Museo Nacional de Arquitectura Palacio de Bell...  
2                                       Santa Teresa  
3                                           Balbuena  
4                                               real

###Funciones de muestreo

####Muestreo aleatorio simple

In [5]:
aleatorio_8 = econdata.sample(n=6)
aleatorio_8

id                  geo_point_2d  \
122  122  19.4328655483,-99.1385210348   
56    56  19.4452114301,-99.1425281852   
22    22  19.4348361174,-99.1467786287   
223  223  19.4285106481,-99.1367967407   
147  147  19.4383757081,-99.1416066037   
94    94  19.4338001466,-99.1463273233   

                                             geo_shape   clave_cat  \
122  {"type": "MultiPoint", "coordinates": [[-99.13...  001_020_04   
56   {"type": "MultiPoint", "coordinates": [[-99.14...  003_043_01   
22   {"type": "MultiPoint", "coordinates": [[-99.14...  002_015_01   
223  {"type": "MultiPoint", "coordinates": [[-99.13...  001_055_10   
147  {"type": "Polygon", "coordinates": [[[-99.1416...  003_098_01   
94   {"type": "MultiPoint", "coordinates": [[-99.14...  002_016_02   

     delegacion perimetro     tipo        nom_id  
122   Cuautémoc         A    Hotel     Principal  
56    Cuautémoc         B    Hotel       Morelia  
22    Cuautémoc         B    Hotel    Fiesta Inn  
223   Cuautémoc         A    Hotel          Niza  
147  Cuauhtémoc         A  Mercado    2 de Abril  
94    Cuautémoc         B    Museo  Arte Popular

####Muestreo sistematico

In [6]:
def muestreo_sistematica(data,pasos):
  indices = np.arange(0,len(data),step=pasos) ##Aqui ponemos que inicie de fila 0 al largo del DF
##La variable step   indica los brincos que ira dando el nuevo DF
  muestra_sistemaca = data.iloc[indices] ##Guarda el nuevo DF con los indices
  return muestra_sistemaca

muestra = muestreo_sistematica(econdata,3) ##Da brincos de 3 en 3
muestra

id                  geo_point_2d  \
0      0   19.424781053,-99.1327537959   
3      3      19.42489472,-99.12073393   
6      6      19.43553422,-99.12324801   
9      9  19.4407152937,-99.1498060057   
12    12      19.43990186,-99.14813347   
..   ...                           ...   
216  216  19.4247697438,-99.1249707246   
219  219  19.4234096295,-99.1438351156   
222  222   19.4397905042,-99.134628735   
225  225      19.43094655,-99.12455418   
228  228  19.4406567734,-99.1394531646   

                                             geo_shape   clave_cat  \
0    {"type": "Polygon", "coordinates": [[[-99.1332...  307_130_11   
3    {"type": "MultiPoint", "coordinates": [[-99.12...  323_102_06   
6    {"type": "MultiPoint", "coordinates": [[-99.12...  318_116_11   
9    {"type": "MultiPoint", "coordinates": [[-99.14...  012_146_22   
12   {"type": "MultiPoint", "coordinates": [[-99.14...  003_079_16   
..                                                 ...         ...   
216  {"type": "Polygon", "coordinates": [[[-99.1250...  323_118_37   
219  {"type": "MultiPoint", "coordinates": [[-99.14...  002_118_03   
222  {"type": "MultiPoint", "coordinates": [[-99.13...  004_063_10   
225  {"type": "MultiPoint", "coordinates": [[-99.12...  323_027_04   
228  {"type": "MultiPoint", "coordinates": [[-99.13...  004_069_01   

              delegacion perimetro     tipo                  nom_id  
0             Cuauhtémoc         B  Mercado             Pino Suárez  
3    Venustiano Carranza         B    Hotel                Balbuena  
6    Venustiano Carranza         B    Hotel    San Antonio Tomatlan  
9              Cuautémoc         B    Hotel                 Detroit  
12             Cuautémoc         B    Hotel                  La Paz  
..                   ...       ...      ...                     ...  
216  Venustiano Carranza         B  Mercado  Lamininas de La Merced  
219            Cuautémoc         B    Hotel          Prado Floresta  
222            Cuautémoc         A    Hotel                  Ladero  
225  Venustiano Carranza         B    Hotel           Gran Veracruz  
228            Cuautémoc         B    Museo        Tequila y Mescal  

[77 rows x 8 columns]

####Muestreo estratificado

In [17]:
econdata['estratificado'] = econdata['delegacion']+","+econdata['tipo']
round((econdata['estratificado'].value_counts()/len(econdata)).sort_values(ascending=False),2)

Cuautémoc,Hotel                0.64
Cuautémoc,Museo                0.16
Venustiano Carranza,Hotel      0.08
Cuauhtémoc,Mercado             0.07
Venustiano Carranza,Mercado    0.05
Name: estratificado, dtype: float64

Los datos confirman la siguiente proporcion aproximada

1)Hoteles en Cuauhtémoc : 0.64

2)Museos en Cuauhtémoc : 0.16

3)Hoteles en Venustiano Carranza : 0.08

4)Mercados en Cuauhtémoc : 0.07

5)Mercados en Venustiano Carranza: 0.05